language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/SpreadFieldCompletionItemBuilder.java

```diff
@@ -0,0 +1,85 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langserver.completions.builder;
+
+import io.ballerina.compiler.api.symbols.FunctionSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import org.apache.commons.lang3.tuple.Pair;
+import org.ballerinalang.langserver.commons.BallerinaCompletionContext;
+import org.eclipse.lsp4j.CompletionItem;
+import org.eclipse.lsp4j.CompletionItemKind;
+
+/**
+ * This class is being used to build spread field completion item.
+ *
+ * @since 2201.0.3
+ */
+public class SpreadFieldCompletionItemBuilder {
+
+    /**
+     * Build the constant {@link CompletionItem}.
+     *
+     * @param symbol   {@link Symbol}
+     * @param typeName type name of the {@link Symbol}
+     * @return {@link CompletionItem} generated completion item
+     */
+    public static CompletionItem build(Symbol symbol, String typeName) {
+        String symbolName = symbol.getName().orElseThrow();
+        String insertText = "..." + symbolName;
+        return build(insertText, insertText, CompletionItemKind.Variable, symbolName, typeName);
+    }
+
+    /**
+     * Build the constant {@link CompletionItem}.
+     *
+     * @param symbol   {@link FunctionSymbol}
+     * @param typeName type name of the {@link Symbol}
+     * @param context  completion context
+     * @return {@link CompletionItem} generated completion item
+     */
+    public static CompletionItem build(FunctionSymbol symbol, String typeName, BallerinaCompletionContext context) {
+        String symbolName = symbol.getName().orElseThrow();
+        Pair<String, String> functionInvocationSignature =
+                FunctionCompletionItemBuilder.getFunctionInvocationSignature(symbol, symbolName, context);
```

Comment:
Avoid using `FunctionCompletionItemBuilder.getFunctionInvocationSignature`. We are not supposed to use the method as a utility method. If there is a need, we have to move it properly to a utility.